tests/test_integration.py

```diff
@@ -46,6 +48,29 @@ def test_pypi_upload(sampleproject_dist):
     cli.dispatch(command)
 
 
+def test_pypi_error(sampleproject_dist, monkeypatch):
+    command = [
+        "twine",
+        "upload",
+        "--repository-url",
+        "https://test.pypi.org/legacy/",
+        "--username",
+        "foo",
+        "--password",
+        "bar",
+        str(sampleproject_dist),
+    ]
+    monkeypatch.setattr(sys, "argv", command)
+
+    message = (
+        "HTTPError from https://test.pypi.org/legacy/: 403 Client Error\n"
+        "Invalid or non-existent authentication information. "
+        "See https://test.pypi.org/help/#invalid-auth for details"
+    )
```

Comment:
This will break if the error message changes (like, when I finish up https://github.com/pypa/warehouse/pull/7808), but I think that's a useful warning that there's been a change in Warehouse that Twine might need to respond to.

Reply:
Can we be fuzzier about this to avoid the tests breaking overnight? Perhaps looking for `403` or `authentication` in the message?

Reply:
I thought about that, but also thought the test breaking would be a useful indication that something has changed on Warehouse that we might want to know about. And, it's an easy test to fix.

But, I'm happy to change it, if you prefer.

Reply:
If we have to write fragile tests to get notified of something changing in Warehouse that might affect us, that only goes to prove that the communication between the two projects is completely broken.